lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -600,6 +600,37 @@ SILCombiner::optimizeConcatenationOfStringLiterals(ApplyInst *AI) {
   return tryToConcatenateStrings(AI, Builder);
 }
 
+/// Determine the pattern for global_addr..
+/// %3 = global_addr @$P : $*SomeP
+/// %4 = init_existential_addr %3 : $*SomeP, $SomeC
+/// %5 = alloc_ref $SomeC
+/// store %5 to %4 : $*SomeC
+static SILValue findInitExistentialFromGlobalAddr(GlobalAddrInst *GAI,
+                  CopyAddrInst *CAI) {
+  /// Check for a single InitExistential.
```

Comment:
There is an assumption in this code that CAI->getSrc() == GAI that is not apparent from the title. Can you at least add the assertion here... maybe something like:

```
assert(CAI->getSrc() == SILValue(GAI) && "Broken Assumption! Global Addr is not the source of the passed in copy_addr?!");
```